### PR TITLE
Update list editing functions

### PR DESCRIPTION
**Pull request by ericyd**
This is an awesome extension, thank you.

One pattern I use frequently is nested ordered lists. This is a simple update to reset the market to 1 when tabbing so that the nested list doesn't continue the numbering of the outer list.

I would like to add a way to resume counting with the outer list when shift-tabbing, but that seems like a substantially more complicated update.

Before
```md
1. list 1
2. list 2
    3. list 2a
    4. list 2b
```
After
```md
1. list1
2. list2
    1. list 2a
    2. list 2b
```

### CODE REVIEW

**Comment by ericyd** [comment]
The failing test in appveyor appears to be unrelated to this update

File: src/listEditing.ts [hunk 1]
```diff
@@ -94,12 +94,27 @@ async function onTabKey() {
...
+            let leadingSpace = matches[1];
+            let delimiter = matches[3];
+            let trailingSpace = " ";
+            let gfmCheckbox = matches[5].replace('[x]', '[ ]');
+
+            const toBeAdded = leadingSpace + marker + delimiter + trailingSpace + gfmCheckbox;
+            await editor.edit(editBuilder => {
+                editBuilder.replace(new Range(new Position(editor.selection.start.line, 0), cursorPos), `${toBeAdded}`);
+            });
+        }
```
Comment (neilsustc) [inline src/listEditing.ts]: It is unnecessary. If you reach this branch, that means it is the case that we are going to indent. So you just need to find the marker and then replace it with `1`

**Comment by neilsustc** [comment]
```markdown
1. test
    1. test
    2. test
    3. |
```

→ pressing <kbd>Backspace</kbd> →

```markdown
1. test
    1. test
    2. test
2. |
<!-- not `3. ` anymore -->
```

**Comment by ericyd** [comment]
OK I'm going to work on this a bit more and try to address [32](https://github.com/neilsustc/vscode-markdown/issues/32) as well.

**Comment by ericyd** [comment]
Is there a way to run the tests locally? I get errors when I run `npm t`, but i would like to add some tests for these updates without having to push here to have AppVeyor validate the updates.

**Comment by neilsustc** [comment]
image

**Comment by neilsustc** [comment]
I note that you are overriding `Alt + Up/Down`. I would suggest to define a function `sortOrderedList()` which takes a `Range` containing the list as input and formats it. Then chain this function after the moving line action. `commands.execute('editor.action.moveLinesUpAction').then(sortOrderedList)`

**Comment by neilsustc** [comment]
Nice work! Since all related tests are passed, I am going to merge it first and make some personal modifications later.

Thanks a lot 🍻

**Comment by ericyd** [comment]
Thanks for the feedback and guidance, and for maintaining this extension.

**Pull request merged**